src/Haskell/Ide/Engine/Transport/LspStdio.hs

```diff
@@ -120,12 +127,18 @@ run dispatcherProc cin _origDir captureFp = flip E.catches handlers $ do
         return Nothing
 
   flip E.finally finalProc $ do
-    CTRL.run (getConfig, dp) (hieHandlers rin) hieOptions captureFp
+    CTRL.run (getConfig, dp) (hieHandlers rin) (hieOptions (BM.elems commandUUIDs)) captureFp
  where
   handlers  = [E.Handler ioExcept, E.Handler someExcept]
   finalProc = L.removeAllHandlers
   ioExcept (e :: E.IOException) = print e >> return 1
   someExcept (e :: E.SomeException) = print e >> return 1
+  getCommandUUIDs = do
+    gen <- getStdGen
+    let commands = ["hare:demote", "applyrefact:applyOne"]
+        uuids :: [UUID]
+        uuids = randoms gen
+    return $ BM.fromList (zip commands uuids)
```

Comment:
I think it might be good to include the original command too, so if the UUID is `abcdef`, then use a command string of `hare:demote:abcdef`.

It will make interpreting logs etc much easier.

Reply:
And then we only need one UUID per session too, come to think of it.